rules/common.smk

```diff
@@ -1,17 +1,45 @@
 import pandas as pd
+from peppy import Project, SAMPLE_NAME_COLNAME as PEP_SAMPLE_COL
 from snakemake.utils import validate
 
-report: "../report/workflow.rst"
+
+def peppy_rename(df):
+    """ Rename peppy's column for sample name identification to snakemake's. """
+    if df is None:
+        return None
+    if "sample" in df.columns and PEP_SAMPLE_COL in df.columns:
+        raise Exception("Multiple sample identifier columns present: {}".format(", ".join(["sample", PEP_SAMPLE_COL])))
+    return df.rename({PEP_SAMPLE_COL: "sample"}, axis=1)
+
+
+def peppy_units(df):
+    """ Add unit/subsample indices to peppy a data frame.  """
+    if "unit" in df.columns:
+        return df
+    def count_names(names):
+        def go(rem, n, curr, acc):
+            if rem == []:
+                return acc + [n]
+            h, t = rem[0], rem[1:]
+            return go(t, n + 1, curr, acc) if h == curr else go(t, 1, h, acc + [n])
+        return go(names[1:], 1, names[0], []) if names else []
+    df.insert(1, "unit", [i for n in count_names(list(df[PEP_SAMPLE_COL])) for i in range(1, n + 1)])
+    return df
```

Comment:
So, the peppy dataframe does contain a column subsample_name, right? So, we can change from unit to subsample_name in the workflow.

Reply:
It does; OK, sounds good

Reply:
we were trying to make it so that existing workflows could work without any updates, but yeah, it would be easier to change the name. my suggestion below of a snakePEP class could also hide this away.